scripts/stats.py

```diff
@@ -182,52 +212,181 @@ def check_search(query, search):
+                    "MISSING",
+                    0,
+                    0,
+                    experiments,
+                    targets,
+                    acquisitions,
+                    0,
+                    0,
+                    0,
+                    0,
+                    None,
+                    None,
+                    "",
+                )
             else:
                 for x in rv:
-                    plate_id, plates, wells, images, planes, bytes = x
-                    plate_count += plates
-                    well_count += wells
-                    image_count += images
-                    if planes:
-                        plane_count += planes
-                    if bytes:
-                        byte_count += bytes
-                    else:
+                    (
+                        plate_or_dataset_id,
+                        plate_or_datasets,
+                        wells,
+                        images,
+                        planes,
+                        bytes,
+                        avg_image_dim
+                    ) = x
+                    if not planes:
+                        planes = 0
+                    if not bytes:
                         bytes = 0
-                    if plates != len(expected):
-                        plates = "%s of %s" % (plates, len(expected))
-                    tb.row(container, plate_id, plates, wells, images, planes,
-                           filesizeformat(bytes))
-    tb.row("Total", "", plate_count, well_count, image_count, plane_count,
-           filesizeformat(byte_count))
-    print(str(tb.build()))
+                    if plate_or_datasets != nexpected:
+                        logging.warning(
+                            '%s: got %d plate/datasets expected %d',
+                            container, plate_or_datasets, nexpected)
+                    if fsusage:
+                        fs_size, fs_num = fs_usage(
+                            client, parenttype, plate_or_dataset_id)
+                    else:
+                        fs_size = bytes
+                        fs_num = None
+                    if fs_num:
+                        fs_avg_size = fs_size / fs_num
+                    else:
+                        fs_avg_size = None
+                    df.loc[len(df)] = (
+                        container1,
+                        container2,
+                        introduced,
+                        plate_or_dataset_id,
+                        nexpected,
+                        wells,
+                        experiments,
+                        targets,
+                        acquisitions,
+                        images,
+                        planes,
+                        fs_size / 2 ** 40,
+                        fs_size,
+                        fs_num,
+                        fs_avg_size,
+                        avg_image_dim,
+                    )
+
+    if append_totals:
```

Comment:
As we are moving away from a recalculation of the stats for all the studies, what is the driver for this sum ? Is it primarily backwards compatibility?

Reply:
Yes, only for backwards compatibility. If there's agreement I'm more than happy to remove it since it'll simplify some of the ouput logic